### PR TITLE
WIP Network operator

### PR DESCRIPTION
**Pull request by ariordan-redhat**
@openshift/team-documentation For peer review, please.

### CODE REVIEW

File: networking/configuring-network-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{product-title} generates the 'Operator configuration' from the 'Cluster
+configuration'.
+
+Once a cluster is running, it is not safe to change most of the network
+settings.
+The Network Operator prevents unsafe changes.
+
+//include::modules/configuration-resource-overview.adoc[leveloffset=+1]
+//
+//include::modules/networking-cluster-configuration.adoc[leveloffset=+1]
```
Comment (vikram-redhat) [inline networking/configuring-network-operator.adoc]: For some reason, even though this is commented out, the build is trying to include it. @ariordan-redhat  did you want to take it out?

**Comment by vikram-redhat** [comment]
@ariordan-redhat  if these PRs have a jira or bug attached, add that as part of the title/comment.

File: modules/nw-modifying-nwoperator-config-startup.adoc [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    additionalNetworks: null
+    clusterNetworks:
+    - cidr: 10.128.0.0/14 <2>
+      hostSubnetLength: 9 <2>
+    defaultNetwork:
+      type: OpenShiftSDN <2>
+    serviceNetwork: 172.30.0.0/16 <2>
+
+----
+
```
Comment (kalexand-rh) [inline modules/nw-modifying-nwoperator-config-startup.adoc]: Remove this extra line.

File: modules/nw-nwop-config-OpenShiftSDN.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      openshiftSDNConfig:
+        mode: NetworkPolicy
+        vxlanPort: 4789
+        mtu: 1450
+        useExternalOpenvswitch: false
+----
++
+The parameter values in the example above are the defaults.
+
+. Edit the parameter values if necessary.
```
Comment (kalexand-rh) [inline modules/nw-nwop-config-OpenShiftSDN.adoc]: You aren't giving me enough information to do this here.

File: modules/nw-nwop-viewing-config.adoc [hunk 4]
```diff
@@ -0,0 +1,94 @@
+// Module filename: nw-nwop-viewing-config.adoc
+// Module included in the following assemblies:
+// * networking/configuring-network-operator.adoc
+
+[id='nw-nwop-viewing-config-{context}']
+= Viewing the Network Configuration in a Running Cluster
+
+
```
Comment (kalexand-rh) [inline modules/nw-nwop-viewing-config.adoc]: You need an intro statement. Why do I want to do this? To confirm that the cluster's using the right network settings?

File: modules/nw-nwop-viewing-config.adoc [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+.Prerequisites
+
+* Start a cluster and login using your credentials.
```
Comment (kalexand-rh) [inline modules/nw-nwop-viewing-config.adoc]: log in

If we're turning this into an installation assembly, you should include the modules to install the oc cli.

File: modules/nw-nwop-viewing-config.adoc [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+.Procedure
+
+. Use the following command to view the `Network.config.openshift.io` CR.
++
+----
+$ oc get Network.config.openshift.io -o yaml
+----
++
+The cluster used in the example output below uses the default configuration values.
```
Comment (kalexand-rh) [inline modules/nw-nwop-viewing-config.adoc]: not below.
> ! below adj
Do not use to indicate a relative location in a document, as in “the information below.” Use “the information
later in this section.”
Alternatively, you can use “following” to refer to a table, to art, or to a list. Example: “The following table
shows ...”

Maybe "The following output contains the default configuration values."

File: modules/nw-nwop-viewing-config.adoc [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
++
+[source,yaml]
+----
+  apiVersion: v1
```
Comment (kalexand-rh) [inline modules/nw-nwop-viewing-config.adoc]: Is this indentation right?

File: modules/nw-nwop-viewing-config.adoc [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+. Use the following command to view the
+`NetworkConfig.networkoperator.openshift.io` CR.
++
+----
+$ oc get NetworkConfig.networkoperator.openshift.io -o yaml
+----
+
+[source,yaml]
+----
+  apiVersion: v1
```
Comment (kalexand-rh) [inline modules/nw-nwop-viewing-config.adoc]: Is this indentation right?

File: modules/nw-nwop-modifying-config-running.adoc [hunk 9]
```diff
@@ -0,0 +1,75 @@
+// Module filename: nw-nwop-modifying-config-running.adoc.adoc
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: This needs to be in a separate assembly. It's a different user story. In the new assembly, I'd reference the "installing {product-title} on AWS with network configurations" assembly as a prereq.

File: modules/nw-nwop-modifying-config-running.adoc [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+// Module included in the following assemblies:
+// * networking/configuring-network-operator.adoc
+
+[id='nw-nwop-modifying-config-running-{context}']
+= Modifying Network Configuration in a Running Cluster
+
+You can only modify `kubeProxy` configuration parameters in
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: s/only modify/modify only

File: modules/nw-nwop-modifying-config-running.adoc [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+a running cluster.
+
+The Network Operator reconciles the state of the cluster
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: Is this helpful here?

File: modules/nw-nwop-modifying-config-running.adoc [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+against a desired configuration.
+
+.Prerequisites
+
+* Login to a running cluster.
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: Log in

Comment (kalexand-rh) [reply modules/nw-nwop-modifying-config-running.adoc]: You also need to have the oc cli installed.

File: modules/nw-nwop-modifying-config-running.adoc [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+.Procedure
+
+. Use the following command to edit the
+`NetworkConfig.networkoperator.openshift.io` Custom Resource (CR).
++
+----
+$ oc edit networkconfig
+----
++
+The example output below shows a `NetworkConfig.networkoperator.openshift.io` CR
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: not below.

File: modules/nw-nwop-modifying-config-running.adoc [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+  status: {}
+
+----
+
+. Add the following code to the `spec` field in the
+`NetworkConfig.networkoperator.openshift.io` CR.
++
+[source,yaml]
+----
+spec:
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: Make sure that the new assembly contains sufficient information about the parameter values.

File: modules/nw-nwop-modifying-config-running.adoc [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+   iptablesSyncPeriod: 30s
+   bindAddress: 0.0.0.0
+   proxyArguments:
+   - "iptables-min-sync-period": ["30s"]
+----
+
+. Save the file and quit the text editor.
++
+The Network Operator checks the syntax when you quit the file. If your
+modifications contain a syntax error, the editor will re-open the file, and display an error
```
Comment (kalexand-rh) [inline modules/nw-nwop-modifying-config-running.adoc]: s/file,/file

**Review by kalexand-rh — COMMENTED** [review]
I think that this assembly needs to be two assemblies, one about installing a cluster with network customizations and one about modifying the networking settings after installation.

The installation assembly should go in the installing/installing_aws folder and also incorporate all of the modules from https://docs.openshift.com/container-platform/4.0/installing/installing_aws/installing-aws-customizations.html. When the user installs a cluster, they'll only use this new assembly, not both the networking and the installation assemblies.

Please let me know if you have questions, want to work on the structure of the new assembly, or want another review of the updated PR. (There are some spots in the current installation assemblies  that will need to change based on the configuration that you're adding.)

File: networking/configuring-network-operator.adoc [hunk 16]
```diff
@@ -0,0 +1,60 @@
+// Assembly filename: configuring-network-operator.adoc
+// Explains network operator, shows default CRDs and modifications to setup.
+[id='configuring-network-operator’]
```
Comment (kalexand-rh) [inline networking/configuring-network-operator.adoc]: I think there's a backtick here, too.

File: networking/configuring-network-operator.adoc [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+include::modules/nw-nwop-config-OpenShiftSDN.adoc[leveloffset=+2]
+
+//Viewing network configuration in a running cluster
+include::modules/nw-nwop-viewing-config.adoc[leveloffset=+1]
+
+//Modifying network configuration in a running cluster
+include::modules/nw-nwop-modifying-config-running.adoc[leveloffset=+1]
+//
+//Full operator config. Network operator uses 2 CRDs, so separated them into 2 docs.
+//include::modules/network-annotated-crd.adoc[leveloffset=+1]
```
Comment (kalexand-rh) [inline networking/configuring-network-operator.adoc]: For each of the commented-out include statements, try replacing "::" with "  ". That seems to make Travis happy.

**Comment by jboxman** [comment]
Shifting working copy to PR https://github.com/openshift/openshift-docs/pull/14224. All fixes will be applied there.

File: modules/nw-modifying-nwoperator-config-startup.adoc [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+This procedure uses the `install-config.yaml` file that was generated
+when you modified the basic network configuration.
+
+.Prerequisites
+
+* You need the `install-config.yaml` file.
+
+.Procedure
+
+. Use the following command to create `manifests`.
```
[the remark this answers is not in the document]
Comment (jboxman) [reply modules/nw-modifying-nwoperator-config-startup.adoc]: @kalexand-rh, this procedure came from the following presentation by Casey Callendrello:

https://docs.google.com/presentation/d/1psQ02UldSc6_7VBc_tbHDV5GsK2CnOaqr20xOtqYbnI/edit#slide=id.g4d6b6a6774_0_86

[the remark this answers is not in the document]
Comment (kalexand-rh) [reply modules/nw-modifying-nwoperator-config-startup.adoc]: @jboxman, as I said, the last I heard was that it was not supported. Here's where QE was discussing it: https://github.com/openshift/openshift-docs/pull/13468#discussion_r252524529

If it is supported, that's fine, but we need to update the installation docs and get QE to confirm them.

**Comment by kalexand-rh** [comment]
Closing since there's a follow-up PR.

**Pull request closed without merging**